src/Orleans/Core/Exceptions.cs

```diff
@@ -207,5 +207,29 @@ protected ClientNotAvailableException(SerializationInfo info, StreamingContext c
         { }
 #endif
     }
+
+    /// <summary>
+    /// Indicates that a <see cref="GrainReference"/> was not bound to the runtime before being used.
+    /// </summary>
+    [Serializable]
+    public class GrainReferenceNotBoundException : OrleansException
```

Comment:
My understanding was that all of our exceptions should inherit from it. Maybe we should have a 'TransientException' base class if there is supposed to be any different behaviour for transient exceptions.